docs/DeveloperGuide.md

```diff
@@ -1,26 +1,237 @@
+
+The Architecture Diagram above provides a high-level view of the design of NUSchedule. The app can be broken down into 
+5 different components:  
+1. Main: Initializes the other components and connects them with each other.
+2. UI: Manages the User Interface that the user interacts with.
+3. Logic: Interprets user commands.
+4. Storage: Reads data from and writes data to the hard disk.
+5. Model: Stores the data the app uses in memory.
+
+__How the architecture components interact with each other__  
+The sequence diagram below shows how each individual component interacts with each other when the user inputs a command.
+![sequence](../diagrams/ArchitectureSequence.png)
+
+### 3.2 UI
+![UI Class Diagram](diagrams/Ui.png)
+
+*Figure 3.2.1 Class diagram for UI component*
+
+ __API__:`UI.java`  
+The UI consists of various parts, e.g. `printGreetingMessage`, `printEventList`, `showError`, `printNumEvent`, etc.
+ 
+The `UI` component reads user input using readCommand() and executes user commands according to the Logic component.
+
+### 3.3 Storage
+__API__:`Storage.java`
+The `Storage` component can save the list of event data in .txt format and read it back.   
+### 3.4 Logic
+__API__:`Parser.java`  
+
+![Logic Class Diagram](diagrams/LogicClass.png)
+
+*Figure 3.4.1 Class diagram for Logic component*
+
+The `logic` component parses the user input and executes commands based on the given input.
+ 1.	Parser reads the user input and returns a command to be executed by `NuSchedule`.
+ 2.	The command execution affects the EventList (e.g. clearing the list).
+ 3.	The result passes back to the UI, which then displays relevant feedback to the user.  
+
+### 3.5 Model
+__API__:`EventList.java`   
+The `Model` component stores an ArrayList, events, that represents the current list of events.
+
+### 3.6 Location
+![Location class diagram](diagrams/Location.jpeg)
+*Figure 3.6.1 Class diagram for location component*
+
+__API:__ `LocationList.java` <br>
+
+The `LocationList` is made up of an ArrayList of Locations, which is a type of variable that stores different information 
+for different types of location (eg. `Building`, `Hostel`, etc.). However, do note that any place that is 
+not within the list of saved locations will be saved as type `OutOfNUS`, and would not be saved into the `location.txt` 
+file. The location will still be part of the location list before the app closes, and it will also be 
+saved as part of the Event information.
+
+<br>There is a separate api for bus stops (`BusStopList.java`), which stores all available bus stops 
+and their information. Although the variable is declared in the main NUScheule class, it is only referenced 
+and not edited in any part of the program.
+
+<br>The `Location` component 
+* stores information about various types of locations 
+* prints the list of locations that is saved in the data file 
+* checks if a location is saved in the list and returns the location when asked
```

Comment:
Does the Location component stores the information about the various types of locations? Sounds like it this is the LocationList component.